c2corg_api/ext/colander_ext.py

```diff
@@ -53,7 +53,10 @@ def deserialize(self, node, cstruct):
         if cstruct is null or cstruct == '':
             return null
         try:
-            return wkbelement_from_geojson(cstruct, self.srid)
+            geom_type = json.loads(cstruct)['type'].upper()
```

Comment:
it is adjusted to load JSON only once but I had to adjust few other tests because I moved json loading function out of `wkbelement_from_geojson`.